Coding/fw/lib/WebServer/Websocket.cpp

```diff
@@ -0,0 +1,131 @@
+/***************************************************************************************************
+  (c) NewTec GmbH 2020   -   www.newtec.de
+  $URL: https://github.com/NewTec-GmbH/esp32-can-iot $
+***************************************************************************************************/
+/**
+@addtogroup Webserver
+@{
+@file       Websocket.cpp
+
+Configuration of ESP32 WebSocket. @ref Websocket.h
+
+* @}
+***************************************************************************************************/
+/* INCLUDES ***************************************************************************************/
+#include "Websocket.h"
+#include <SPIFFS.h>
+
+/* C-Interface ************************************************************************************/
+extern "C"
+{
+}
+
+/* CONSTANTS **************************************************************************************/
+AsyncWebSocket ws("/ws");
+
+/* MACROS *****************************************************************************************/
+
+/* TYPES ******************************************************************************************/
+
+/* PROTOTYPES *************************************************************************************/
+static void handleWebSocketMessage(void *arg, uint8_t *data, size_t len);
+static void onEvent(AsyncWebSocket *server, AsyncWebSocketClient *client, AwsEventType type,
+                    void *arg, uint8_t *data, size_t len);
+
+/* VARIABLES **************************************************************************************/
+
+static QueueHandle_t inputQueue;
+
+/* PUBLIC METHODES ********************************************************************************/
+
+/**************************************************************************************************/
+
+/*
+*   Initializes the WebSocket Service
+*/
+void websocket::init(AsyncWebServer &server)
+{
+    server.on("/communication", HTTP_GET, [](AsyncWebServerRequest *request) {
+        request->send(SPIFFS, "/ws.html");
+    });
+
+    ws.onEvent(onEvent);
+    server.addHandler(&ws);
+    inputQueue = xQueueCreate(100, sizeof(char));
+    if(NULL == inputQueue)
+    {
+        Serial.println("Error at creating the Queue");
+    }
+}
+
+/**************************************************************************************************/
+
+/*
+*   Send WebSocket Message
+*/
+void websocket::send(const String &message)
+{
+    String systime = String(millis());
+    ws.textAll(message + systime);
+}
+
+/**************************************************************************************************/
+
+/*
+*   Receive WebSocket Message
+*/
+bool websocket::receive(char &c)
+{
+    return xQueueReceive(inputQueue, &c, 100);;
```

Comment:
is it really ok to wait 100 ticks for data ?  I would have expected 0 to intantly return false if no input.

return should be "return xQueueReceive(...) == pdTRUE;"